packages/formspree-react/src/useForm.ts

```diff
@@ -67,144 +41,36 @@ const useForm = <T extends FieldValues>(
     );
   }
 
-  if (formspreeContext.client && formspreeContext.client.stripePromise) {
-    // suppress existing violation: React Hook "useStripe" is called conditionally
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    stripe = useStripe();
-    // suppress existing violation: React Hook "useElements" is called conditionally
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    elements = useElements();
-  }
-
-  const extraData = args.data;
-
-  const reset: ResetFunction = () => {
-    setErrors(null);
-    setResult(null);
-    setSubmitting(false);
-    setSucceeded(false);
-  };
-
-  const handleSubmit: SubmitHandler<T> = async (submissionData) => {
-    const getFormData = (event: FormEvent) => {
-      event.preventDefault();
-
-      const form = event.target as HTMLFormElement;
-      if (form.tagName != 'FORM') {
-        throw new Error('submit was triggered for a non-form element');
-      }
-      return new FormData(form);
-    };
-
-    const formData = isEvent(submissionData)
-      ? getFormData(submissionData)
-      : submissionData;
-
-    // Append extra data from config
-    if (typeof extraData === 'object') {
-      for (const [prop, value] of Object.entries(extraData)) {
-        let extraDataValue: string | undefined;
-        if (typeof value === 'function') {
-          extraDataValue = await value();
-        } else {
-          extraDataValue = value;
-        }
-        if (extraDataValue !== undefined) {
-          appendExtraData(formData, prop, extraDataValue);
-        }
-      }
-    }
-
-    const createPaymentMethod = async () => {
-      // @ts-ignore: unhandled stripe is possibly null
-      const payload = await stripe.createPaymentMethod({
-        type: 'card',
-        // @ts-ignore: unhandled elements is possibly null and getElement can return null
-        card: elements.getElement(CardElement),
-        billing_details: mapBillingDetailsPayload(formData),
-      });
-
-      return payload;
-    };
-
-    setSubmitting(true);
-
-    const result = await formspreeContext.client.submitForm(formKey, formData, {
-      endpoint: args.endpoint,
-      clientName: `@formspree/react@${version}`,
-      createPaymentMethod:
-        formspreeContext.client && formspreeContext.client.stripePromise
-          ? createPaymentMethod
-          : undefined,
-    });
-
-    if (isSubmissionError(result)) {
-      setErrors(result);
-      setResult(null);
+  const submit = useSubmit<T>(formKey, {
+    client: args.client,
+    extraData: args.data,
+    onError(error) {
+      setErrors(error);
+      setSubmitting(false);
       setSucceeded(false);
-    } else {
+    },
+    onSuccess(result) {
       setErrors(null);
       setResult(result);
+      setSubmitting(false);
       setSucceeded(true);
-    }
+    },
+    origin: args.endpoint,
```

Comment:
I think naming this `origin` makes more sense. This is only for the new `useSubmit` hook. We will update the option for `useForm` when we do the major release.